lab_2_tokenize_by_bpe/main.py

```diff
@@ -58,7 +102,24 @@ def train(
     :param num_merges: required number of new tokens
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
-
+    if not isinstance(word_frequencies, dict) or not isinstance(num_merges, int):
+        return None
+    while num_merges > 0:
+        num_of_pairs = count_tokens_pairs(word_frequencies)
+        if num_of_pairs is None:
+            return None
+        if num_merges > len(num_of_pairs):
+            num_merges = len(num_of_pairs)
+        max_freq = max(num_of_pairs.values())
+        pairs_list = []
+        for pair, frequency in num_of_pairs.items():
+            if frequency == max_freq:
```

Comment:
What if several pairs have the same maximum frequency?

Reply:
I tried to change it

Reply:
But I tried to add all pairs with such frequency to the list